src/main/java/edu/cornell/kfs/rass/batch/service/impl/RassSortServiceImpl.java

```diff
@@ -0,0 +1,116 @@
+package edu.cornell.kfs.rass.batch.service.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import edu.cornell.kfs.rass.batch.service.RassSortService;
+import edu.cornell.kfs.rass.batch.xml.RassXmlAgencyEntry;
+
+public class RassSortServiceImpl implements RassSortService {
+    private static final Logger LOG = LogManager.getLogger(RassSortServiceImpl.class);
+
+    @Override
+    public List<RassXmlAgencyEntry> sortRassXmlAgencyEntriesForUpdate(List<RassXmlAgencyEntry> agencies) {
+        AgenciesSortHelper agencySortHelper = new AgenciesSortHelper();
+        sortAgenciesWithNoReportsToAgency(agencies, agencySortHelper);
+        sortAgenciesWithReportsToAgencyThatHasBeenSorted(agencySortHelper);
+        sortLeftOverAgencies(agencySortHelper);
```

Comment:
Are the left over agencies: agencies with reports to agency that has not been sorted? would it make sense to update the method name to be more descriptive?